src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/RollupRunner.kt

```diff
@@ -362,7 +382,16 @@ object RollupRunner : ScheduledJobRunner,
     private suspend fun updateRollupJob(job: Rollup, metadata: RollupMetadata): RollupJobResult {
         try {
             val req = IndexRollupRequest(rollup = job, refreshPolicy = WriteRequest.RefreshPolicy.IMMEDIATE)
-            val res: IndexRollupResponse = client.suspendUntil { execute(IndexRollupAction.INSTANCE, req, it) }
+            val roles = if (job.user == null) {
+                settings.getAsList("", listOf("all_access"))
+            } else {
+                job.user.roles
+            }
+            val res = withCloseableContext(InjectorContextElement(job.id, settings, threadPool.threadContext, roles, job.user?.name)) {
```

Comment:
If a job.user can be null (assuming from before this PR is introduced?) then won't null be passed in to the InjectorContextElement which then means the IndexRollupAction thinks its coming from an external request instead of internal? See above comment, let's not use "user" here for this. Prefer being explicit with an actual field for this behavior.

Reply:
introduce explicit flag for this `INDEX_MANAGEMENT_PLUGIN_INTERNAL`